### PR TITLE
Added extractor API including tests and example code

### PR DESCRIPTION
**Pull request by tifroz**
Did my best fix all the inconsistencies I could find.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,6 +116,24 @@ youtubedl.getSubs(url, options, function(err, files) {
 
 For more usage info on youtube-dl and the arguments you can pass to it, do `youtube-dl -h` or go to the [youtube-dl documentation][].
 
+## Getting the list of extractors
```
Comment (fent) [inline README.md]: new line after heading

File: README.md [hunk 2]
```diff
@@ -116,6 +116,24 @@ youtubedl.getSubs(url, options, function(err, files) {
...
+```js
+var youtubedl = require('youtube-dl');
+youtubedl.getExtractors(true, function(err, list) {
+	console.log('Found ' + list.length + ' extractors');
+	for (var i = 0; i < list.length; i++) {
+		console.log(list[i]);
+	}
```
Comment (fent) [inline README.md]: 2 space tabs are used throughout the module.

File: example/extractors.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+var ytdl = require('..');
+
+ytdl.getExtractors(true, function(err, list) {
+  console.log('Found '+list.length+' extractors');
```
Comment (fent) [inline example/extractors.js]: spaces around operators

File: example/extractors.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  var show = 4;
+  for (var  i = 0; i < Math.min(show,list.length); i++) {
+    console.log(list[i]);
+  }
+  if (list.length > show) {
+    console.log('...' + (list.length-show) + ' not shown')
```
Comment (fent) [inline example/extractors.js]: spaces around operators

File: example/extractors.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+var ytdl = require('..');
+
+ytdl.getExtractors(true, function(err, list) {
+  console.log('Found '+list.length+' extractors');
+  var show = 4;
+  for (var  i = 0; i < Math.min(show,list.length); i++) {
```
Comment (fent) [inline example/extractors.js]: space after comma

File: lib/youtube-dl.js [hunk 6]
```diff
@@ -305,3 +305,21 @@ ytdl.getSubs = function(url, options, callback) {
     callback(null, files);
   });
 };
+
+/**
+ * @params {Boolean} descriptions
+ * @param {Function(!Error, Object)} callback
+ */
+ytdl.getExtractors = function(descriptions, options, callback) {
```
Comment (fent) [inline lib/youtube-dl.js]: should the `options` argument be there? I'm not sure that there are any other options that can be used in youtube-dl when extractors are listed.

**Comment by fent** [comment]
Thanks for making all those changes. I commented, mostly things with styles. Otherwise, most of it looks good.

Also, you can force push commits so that you don't have to recreate the PR after you squash. Not sure if you knew this, but it helps keeping things more organized.

File: lib/youtube-dl.js (hunk 6 above)
Comment (tifroz) [inline lib/youtube-dl.js]: These are the child process execution options, it always makes sense to have them here IMHO - if only for the timeout option (http://nodejs.org/api/child_process.html#child_process_child_process_exec_command_options_callback)

**Comment by tifroz** [comment]
After pushing the updates to my repository, I just hit the 'Edit' then 'Save' buttons on this PR. It seems like that was enough for the PR to be updated with the adjustments but let me know if that wasn't the case

File: lib/youtube-dl.js (hunk 6 above)
Comment (fent) [inline lib/youtube-dl.js]: oh you're right. sorry, for some reason I thought they were args to the command.

File: lib/youtube-dl.js [hunk 7]
```diff
@@ -305,3 +305,21 @@ ytdl.getSubs = function(url, options, callback) {
     callback(null, files);
   });
 };
+
+/**
+ * @params {Boolean} descriptions
```
Comment (fent) [inline lib/youtube-dl.js]: `@param` instead of `@params`

can you also add one in for `options`?

Comment (tifroz) [inline lib/youtube-dl.js]: Good point. Done!

File: lib/youtube-dl.js [hunk 8]
```diff
@@ -305,3 +305,22 @@ ytdl.getSubs = function(url, options, callback) {
     callback(null, files);
   });
 };
+
+/**
+ * @param {Boolean} descriptions
+ * @param {Object} options
```
Comment (fent) [inline lib/youtube-dl.js]: forgot to mention to include a `!` to denote that these two arguments (`descriptions` and `options`) are optional

like [this](https://github.com/fent/node-youtube-dl/blob/master/lib/youtube-dl.js#L33)

**Comment by fent** [comment]
Thank you!

**Comment by tifroz** [comment]
Thank you back for the quick npm publish :)

**Pull request merged**